crates/snapbox/src/data.rs

```diff
@@ -263,6 +362,8 @@ impl Normalize for NormalizeMatches<'_> {
                     .normalize(&text, &self.pattern.render().unwrap());
                 Data::text(lines)
             }
+            #[cfg(feature = "json")]
+            DataInner::Json(_) => todo!("unsure of how to do matches here"),
```

Comment:
Adding a comment to track resolving of this

Reply:
The big question here is what does `NormalizeMatches` mean for structured data?

Is it the exact same as `Text` or something completely different

Reply:
- We should run the text version of `NormalizeMatches` on the strings
- A string that is exactly `"{...}"` matches all data structures
  - In a list, should it also match for N number of elements?
  - Or rather, double check what cargo does for this syntax and we can use that for a conversation starter